Data Structures/Queues/circularqueue.java

```diff
@@ -0,0 +1,99 @@
+import java.io.*;
+import java.util.*;
+
+class cqueue
+{
+	int f=-1,r=-1; //here f is front pointer and r is rear pointer.
+	Scanner x=new Scanner(System.in);
+	int queue[]=new int[5]; //maximum size of queue is 5.
+	public void insert()
+	{
+		int e;
+		if((f==0 && r==4)||((r+1)==f))
```

Comment:
With this condition I can only put 4 elements in the queue. Even though it has a capacity of 5 elements.